packages/kit/test/apps/basics/test/cross-platform/client.test.js

```diff
@@ -881,6 +881,8 @@ test.describe('Load', () => {
 test.describe('Vite', () => {
 	if (process.env.DEV) {
 		test('optimizes dependencies', async ({ page }) => {
+			// wait for Vite to optimize dependencies (otherwise the test fails for Ubuntu firefox)
+			await page.waitForTimeout(1000);
```

Comment:
this is a bit unstable, can we wait for a specific locator or a console log statement?
Another option might be to explicitly list all required deps in optimizeDeps.include here so its eagerly done.

Reply:
Doesn't vite send a reload message over ws when that happens? Can we capture that somehow?

Reply:
there is the connected message you can wait for https://github.com/sveltejs/vite-plugin-svelte/blob/f4d51132e1ea7ce59ef0c99e15716c0248053c30/packages/e2e-tests/vitestSetup.ts#L218  but thats just the client. I don't think that waits until all deps are optimized, esp not for deps that are not discovered during scan. Explicitly listing deps that need optimization could speed up the test too, so i'd try that first.
(unless the scan phase taking a bit is part of the test)

Reply:
The scan phase isn't a part of the test. However, I don't think there are any client packages that still need to be optimised (or are there?)

https://github.com/sveltejs/kit/blob/32c0acc373bfcd22ef1c544af25950a91c5c30a9/packages/kit/test/apps/basics/package.json#L16-L25

https://github.com/sveltejs/kit/blob/32c0acc373bfcd22ef1c544af25950a91c5c30a9/packages/kit/test/apps/basics/vite.config.js#L13

Reply:
that could be a different issue then, unless there is logging that says the optimizer reloaded it. i'll check on linux with ff.

Reply:
Actually I think it's because I added the load event listener to early so maybe it counts the initial page load navigation.